src/asset-management/EulerV2Manager.sol

```diff
@@ -198,14 +152,14 @@ contract AaveManager is IAssetManager, Owned(msg.sender), ReentrancyGuard {
         IERC20 lToken0 = aPair.token0();
         IERC20 lToken1 = aPair.token1();
 
-        IERC20 lToken0AToken = _getATokenAddress(lToken0);
-        IERC20 lToken1AToken = _getATokenAddress(lToken1);
+        IERC4626 lToken0Vault = assetVault[lToken0];
+        IERC4626 lToken1Vault = assetVault[lToken1];
 
-        // do not do anything if there isn't a market for the token
-        if (address(lToken0AToken) == address(0)) {
+        // do not do anything if there isn't a designated vault for the token
+        if (address(lToken0Vault) == address(0)) {
             aAmount0Change = 0;
         }
-        if (address(lToken1AToken) == address(0)) {
+        if (address(lToken1Vault) == address(0)) {
```

Comment:
I checked and it looks like it is but just wanna confirm that setting the delta to zero as there is no change is safe for the callers? For `adjustManagement` and `returnAsset` we could just use `require` right? Then if we made `_calculateChangeAmount` set amount to 0 when there is no manager that wouldnt need this as well?

I guess then we would have two map lookups (though on a warm slot), not sure how bad that is.

What do you think about require vs modifying argument to 0.

Reply:
in the current impl, setting those amounts to 0 is safe for the callers -- the caller won't get a revert. 

Long time ago when I was designing this, I had this consideration too. 

So I think I decided that validating this in `_adjustManagement` is more correct, because it's checking where it matters (here in the core function), right before it interacts with the vaults. If we do it in `adjustManagement` and `returnAsset` it's also correct but it's doing it in the periphery, so to speak. And one day if we introduce another function but forget to check for these conditions it could be problematic. 

In other words, to be robust, we need to validate the inputs in the heart, i.e. `_adjustManagement`. If we do it in the external functions, I think it's at best duplicating the work and at worst opening up to vulnerabilities.

Reply:
cuz `_calculateChangeAmount` used by `afterLiquidityEvent` has no knowledge of whether a certain token has an assigned vault or not. 

And I rather let it do some extra math than do an extra sload to check for the vault. Granted, this sload slot will be warm, but still at least 100 gas. I haven't measured it but my guesstimation is that a few arithmetic operations should still be < 100 gas

Reply:
Sounds good